### PR TITLE
IDENTITY-6450: Update playground with implicit flow for OpenID Connect

### PR DESCRIPTION
**Pull request by piraveena**
@darshanasbg @madurangasiriwardena

### CODE REVIEW

File: modules/samples/oauth2/playground2/src/main/webapp/oauth2-authorize-user.jsp [hunk 1]
```diff
@@ -71,7 +74,18 @@
...
 
     OAuthPKCEAuthenticationRequestBuilder oAuthPKCEAuthenticationRequestBuilder = new OAuthPKCEAuthenticationRequestBuilder(authzEndpoint);
-    if (authzGrantType.equals(OAuth2Constants.OAUTH2_GRANT_TYPE_CODE) && usePKCE) {
+    if (authzGrantType.equals(OAuth2Constants.OAUTH2_GRANT_TYPE_IMPLICIT)) {
+        if (scope.equals(OAuth2Constants.SCOPE_OPENID)) {
+            if (implicitRespType.equals(OpenIDConnectConstants.ID_TOKEN) ||
+                    implicitRespType.equals(OpenIDConnectConstants.ID_TOKEN_TOKEN)) {
+                authzGrantType = implicitRespType;
+                oAuthPKCEAuthenticationRequestBuilder.setParameter(OpenIDConnectConstants.NONCE,
+                        (String) UUID.randomUUID().toString());
```
Comment (omindu) [inline modules/samples/oauth2/playground2/src/main/webapp/oauth2-authorize-user.jsp]: Why cast to String ?

File: modules/samples/oauth2/playground2/src/main/webapp/oauth2-authorize-user.jsp [hunk 2]
```diff
@@ -29,6 +31,7 @@
         String authzGrantType = request.getParameter(OAuth2Constants.OAUTH2_GRANT_TYPE);
         String scope = request.getParameter(OAuth2Constants.SCOPE);
         String callBackUrl = request.getParameter(OAuth2Constants.CALL_BACK_URL);
+        String implicitRespType=request.getParameter(OpenIDConnectConstants.IMPLICIT_RESPONSE_TYPE);
```
Comment (mefarazath) [inline modules/samples/oauth2/playground2/src/main/webapp/oauth2-authorize-user.jsp]: formatting,
ie.String implicitRespType=request. --> String implicitRespType = request

File: modules/samples/oauth2/playground2/src/main/webapp/oauth2-authorize-user.jsp [hunk 3]
```diff
@@ -71,7 +74,18 @@
...
+            if (implicitRespType.equals(OpenIDConnectConstants.ID_TOKEN) ||
+                    implicitRespType.equals(OpenIDConnectConstants.ID_TOKEN_TOKEN)) {
+                authzGrantType = implicitRespType;
+                oAuthPKCEAuthenticationRequestBuilder.setParameter(OpenIDConnectConstants.NONCE,
+                        (String) UUID.randomUUID().toString());
+                session.setAttribute(OpenIDConnectConstants.IMPLICIT_RESPONSE_TYPE, implicitRespType);
+            }
+        }
+    }
+    else if ((authzGrantType.equals(OAuth2Constants.OAUTH2_GRANT_TYPE_CODE) && usePKCE)) {
```
Comment (mefarazath) [inline modules/samples/oauth2/playground2/src/main/webapp/oauth2-authorize-user.jsp]: Move `else if` to line 87

File: modules/samples/oauth2/playground2/src/main/webapp/oauth2.jsp [hunk 4]
```diff
@@ -184,6 +192,43 @@
...
+                var sParam = arrParams[i].split("=");
+
+                if (sParam[0] == "id_token") {
+                    var idToken = sParam[1];
+                    return idToken;
+                }
+            }
+            return "";
+        }
+        function getDecodedIDToken() {
```
Comment (omindu) [inline modules/samples/oauth2/playground2/src/main/webapp/oauth2.jsp]: Can move the duplicated logic to a new function

File: modules/samples/oauth2/playground2/src/main/webapp/oauth2.jsp [hunk 5]
```diff
@@ -47,6 +49,7 @@
 
         error = request.getParameter(OAuth2Constants.ERROR);
         grantType = (String) session.getAttribute(OAuth2Constants.OAUTH2_GRANT_TYPE);
+        implicit_response_type = (String) session.getAttribute(OpenIDConnectConstants.IMPLICIT_RESPONSE_TYPE);
```
Comment (mefarazath) [inline modules/samples/oauth2/playground2/src/main/webapp/oauth2.jsp]: Use camel case for variable names
ie. implicit_response_type --> implicitResponseType

File: modules/samples/oauth2/playground2/src/main/webapp/oauth2.jsp [hunk 6]
```diff
@@ -184,6 +192,43 @@
...
+                var sParam = arrParams[i].split("=");
+
+                if (sParam[0] == "id_token") {
+                    var decodedString = atob(sParam[1].split(".")[1]);
+                    return decodedString;
+                }
+            }
+            return "";
+        }
+        function makeList(data) {
```
Comment (mefarazath) [inline modules/samples/oauth2/playground2/src/main/webapp/oauth2.jsp]: IMO would better to have a new line between two functions

File: modules/samples/oauth2/playground2/src/main/webapp/oauth2.jsp [hunk 7]
```diff
@@ -480,7 +538,60 @@
...
+                    <tr>
+                        <script type="text/javascript">
+                            var decodedIdToken = JSON.parse(getDecodedIDToken());
+                            makeList(decodedIdToken);
+                        </script>
+                    </tr>
+                    </tbody>
+                </table>
+            </div>
+            <% } else if (implicit_response_type != null &&
```
Comment (mefarazath) [inline modules/samples/oauth2/playground2/src/main/webapp/oauth2.jsp]: Can't we simplify 
`implicit_response_type != null && OpenIDConnectConstants.ID_TOKEN_TOKEN.equals(implicit_response_type)
`to
`OpenIDConnectConstants.ID_TOKEN_TOKEN.equals(implicit_response_type)`

**Comment by piraveena** [comment]
Fixed with requested changes

File: modules/samples/oauth2/playground2/src/main/webapp/oauth2.jsp [hunk 8]
```diff
@@ -185,6 +193,41 @@
             return "";
         }
 
+        function getIDtoken() {
+            var fragment = window.location.hash.substring(1);
+            var arrParams = fragment.split("&");
+            for (var i = 0; i < arrParams.length; i++) {
+                var sParam = arrParams[i].split("=");
```
Comment (mefarazath) [inline modules/samples/oauth2/playground2/src/main/webapp/oauth2.jsp]: shall we use a better variable name for sParam?

**Comment by omindu** [comment]
I tried this out and came across an issue. After trying the implicit grant and obtaining the token, if you go to oauth2.jsp in the same browser session, it always shows the below page.

image

**Comment by piraveena** [comment]
I will check this

**Comment by piraveena** [comment]
Clears the current session of the app without showing token info in a new oauth2.jsp page

File: modules/samples/oauth2/playground2/src/main/org/wso2/sample/identity/oauth2/OpenIDConnectConstants.java [hunk 9]
```diff
@@ -0,0 +1,14 @@
+package org.wso2.sample.identity.oauth2;
+
+
```
Comment (Abilashini) [inline modules/samples/oauth2/playground2/src/main/org/wso2/sample/identity/oauth2/OpenIDConnectConstants.java]: Please remove the additional line

File: modules/samples/oauth2/playground2/src/main/org/wso2/sample/identity/oauth2/OpenIDConnectConstants.java [hunk 10]
```diff
@@ -0,0 +1,14 @@
+package org.wso2.sample.identity.oauth2;
```
Comment (Abilashini) [inline modules/samples/oauth2/playground2/src/main/org/wso2/sample/identity/oauth2/OpenIDConnectConstants.java]: Missing license header

**Pull request merged**